delta-app/src/main/java/io/cdap/transformation/TransformationUtil.java

```diff
@@ -107,29 +139,35 @@ public static DDLEvent transformDDLEvent(DDLEvent ddlEvent, List<Transformation>
    * @param table the table that transformations are applied on
    * @return list of transformations applied on the specified table
    */
-  public static List<Transformation> loadTransformations(Configurer configurer,
+  public static Map<String, List<Transformation>> loadTransformations(Configurer configurer,
                                                          Map<String, TableTransformation> tableTransformations,
-                                                         SourceTable table) {
+                                                         SourceTable table) throws TransformationException {
     String tableName = table.getSchema() == null ? table.getTable() : table.getSchema() + "." + table.getTable();
-    List<Transformation> columnTransformations = new ArrayList<>();
+    Map<String, List<Transformation>> columnTransformations = new LinkedHashMap<>();
     TableTransformation tableTransformation = tableTransformations.get(tableName);
     if (tableTransformation == null) {
-      return Collections.emptyList();
+      return Collections.emptyMap();
     }
-    tableTransformation.getColumnTransformations().forEach(t -> {
+    for (ColumnTransformation t : tableTransformation.getColumnTransformations()) {
+      String columnName = t.getColumnName();
       String directive = t.getDirective();
       String directiveName = TransformationUtil.parseDirectiveName(directive);
       try {
         Transformation transformation = configurer.usePlugin(Transformation.PLUGIN_TYPE, directiveName,
                                                              UUID.randomUUID().toString(),
                                                              PluginProperties.builder().build());
         transformation.initialize(new DefaultTransformationContext(directive));
-        columnTransformations.add(transformation);
+
+        if (columnTransformations.containsKey(columnName)) {
+          columnTransformations.getOrDefault(columnName, new ArrayList<>()).add(transformation);
+        } else {
+          columnTransformations.put(columnName, new ArrayList<>(Arrays.asList(transformation)));
+        }
       } catch (Exception e) {
-        throw new RuntimeException(String.format("Failed to load transformation plugin for directive : %s. Error : %s.",
-                                                 directive, e.getMessage()), e);
+        throw new TransformationException(tableName, columnName, String.format("Exception while loading plugin " +
```

Comment:
Better to preserve `e` here. Also log it as DEBUG from wherever its getting called in the end before returning to the user.